generic_ecs/main.tf

```diff
@@ -0,0 +1,52 @@
+resource "aws_ecs_cluster" "ecs_cluster" {
+  name = var.cluster_name
+
+  tags = merge(
+    var.common_tags,
+    tomap(
+      { "Name" = var.cluster_name }
+    )
+  )
+}
+
+resource "aws_ecs_task_definition" "ecs_task" {
+  family                   = var.task_family_name
+  execution_role_arn       = var.execution_role
+  network_mode             = "awsvpc"
+  requires_compatibilities = var.compatibilities
+  cpu                      = var.cpu
+  memory                   = var.memory
+  container_definitions    = var.container_definition
+  task_role_arn            = var.task_role
+
+  tags = merge(
+    var.common_tags,
+    tomap(
+      { "Name" = var.task_family_name }
+    )
+  )
+}
+
+resource "aws_ecs_service" "ecs_service" {
+  name                              = var.service_name
+  cluster                           = aws_ecs_cluster.ecs_cluster.id
+  task_definition                   = aws_ecs_task_definition.ecs_task.arn
+  desired_count                     = 1
+  launch_type                       = "FARGATE"
+  health_check_grace_period_seconds = "360"
```

Comment:
Is it worth having the `health_check_grace_period_seconds` value as a variable, in case different ECS need to be different?

Reply:
Yeah that makes sense. I've moved desired_count into a variable as well, both with these defaults as you might want to change that as well.